### PR TITLE
Remove the pre-command-hook in pre-exit, lest null pointer exception.

### PR DESCRIPTION
**Pull request by dickmao**

### CODE REVIEW

**Comment by tarsius** [comment]
This package is part of Emacs. Have you signed the FSF copyright assignment?

File: lisp/transient.el [hunk 1]
```diff
@@ -1989,20 +1989,17 @@ value.  Otherwise return CHILDREN as is."
...
+      (if-let ((fn (transient--get-predicate-for this-original-command)))
+          (when (eq transient--exit (funcall fn))
+            (setq transient--exitp (or transient--exitp t))
+            (transient--pre-exit))
+        (if (let ((keys (this-command-keys-vector)))
+              (eq (aref keys (1- (length keys))) ?\C-g))
+            (setq this-command 'transient-noop)
+          (unless (transient--edebug-command-p)
+            (setq this-command 'transient-undefined)))
+        (when (eq transient--exit transient--stay)
```
Comment (tarsius) [inline lisp/transient.el:2001]: You are comparing two constants. The result is always `nil`.

**Comment by tarsius** [comment]
> Remove the pre-command-hook in pre-exit, lest null pointer exception.

Please provide more details such as what happened and how to reproduce that.

File: lisp/transient.el (hunk 1 above)
Comment (dickmao) [reply lisp/transient.el]: Word.

**Comment by dickmao** [comment]
> This package is part of Emacs. Have you signed the FSF copyright assignment?

My first answer is money enforces IP and the FSF alas has none.  My second answer is "yes."

**Comment by dickmao** [comment]
> Please provide more details such as what happened and how to reproduce that.

I got this while generating a patch using magit-patch-changelog.
```
  signal(wrong-type-argument ((or eieio-object class cl-structure-object) nil obj))
  eieio-oref(nil transient-non-suffix)
  transient--get-predicate-for(magit-patch-changelog-create)
  transient--pre-command()
  run-hooks(pre-command-hook)
  server-visit-files((("/home/dick/gnus/.git/COMMIT_EDITMSG")) #<process server <1>> nil)
  server-process-filter(#<process server <1>> "-dir /home/dick/gnus/ -current-frame -tty /dev/pts...")
  accept-process-output(nil 0.1)
  magit-patch-changelog-create(nil nil)
```

**Comment by tarsius** [comment]
The `remove-hook` could be moved like this but I do not wish to do so. Transient is complex and little things, like always dealing with the pre and post hooks together, help. Beside this change only helps because your command exits the transient; if it didn't then we could not remove the hook early on. I don't see this as a bug in transient but as a accumulation of unusual things happening. You can deal with this situation by removing the hook explicitly in your command.

Off topic: I noticed you use `magit-section--backward-find`, which was removed a little while ago. If you want to stick to how you currently implement things, then you'll have to copy the definition of that and `magit-section--backward-protected` into your package. I do not understand why you do all the things you do in `magit-patch-changelog-create`, but get a feeling that it would be better to do it on `git-commit-setup-hook`. It appears you parse the process buffer to detect when the commit process has finished. Instead you should store `magit-this-process` right after invoking that command, and then wait for its status to change.

**Pull request closed without merging**